core/src/main/java/org/mule/routing/filters/WildcardFilter.java

```diff
@@ -90,25 +89,34 @@ public boolean accept(Object object)
                     candidate = candidate.toLowerCase();
                 }
 
-                int i = pattern.indexOf('*');
-                if (i == -1)
+                int firstWildcardIndex = pattern.indexOf('*');
+                if (firstWildcardIndex == -1)
                 {
                     foundMatch = pattern.equals(candidate);
                 }
                 else
                 {
-                    int i2 = pattern.indexOf('*', i + 1);
-                    if (i2 > 1)
+                    int secondWildcardIndex = pattern.indexOf('*', firstWildcardIndex + 1);
+                    if (firstWildcardIndex == 0 && secondWildcardIndex == pattern.length()-1)
                     {
-                        foundMatch = candidate.indexOf(pattern.substring(1, i2)) > -1;
+                        //enclosing string case
+                        foundMatch = candidate.indexOf(pattern.substring(1, secondWildcardIndex)) > -1;
                     }
-                    else if (i == 0)
+                    else if (firstWildcardIndex == 0 && secondWildcardIndex == -1)
                     {
+                        //prefix case
                         foundMatch = candidate.endsWith(pattern.substring(1));
                     }
+                    else if (firstWildcardIndex == pattern.length()-1)
+                    {
+                        //suffix case
+                        foundMatch = candidate.startsWith(pattern.substring(0, firstWildcardIndex));
+                    }
                     else
                     {
-                        foundMatch = candidate.startsWith(pattern.substring(0, i));
+                        //no other cases are supported
+                        logger.warn(String.format("wildcard-filter only supports wildcards as prefix (*.log), suffix (java.util.*) or enclosing strings (*util*). Consider using a regex-filter instead for pattern: %s.", pattern));
```

Comment:
I've added the changes for this and the comment below here https://github.com/mulesoft/mule/tree/MULE-9501-refactor-to-consider, but as the commit message says, the change is too risky and expensive considering the uses of WildcardFilter listed in https://www.mulesoft.org/jira/browse/MULE-9501.